torchsde/_core/adjoint.py

```diff
@@ -195,32 +230,62 @@ def sdeint_adjoint(sde: nn.Module,
                          'can be specified explicitly via the `adjoint_params` argument. If there are no parameters '
                          'then it is allowable to set `adjoint_params=()`.')
 
-    sde, y0, ts, bm, method = sdeint.check_contract(sde, y0, ts, bm, method, names, logqp)
+    sde, y0, ts, bm, method, options = sdeint.check_contract(sde, y0, ts, bm, method, adaptive, options, names, logqp)
     misc.assert_no_grad(['ts', 'dt', 'rtol', 'adjoint_rtol', 'atol', 'adjoint_atol', 'dt_min'],
                         [ts, dt, rtol, adjoint_rtol, atol, adjoint_atol, dt_min])
     adjoint_params = tuple(sde.parameters()) if adjoint_params is None else tuple(adjoint_params)
     adjoint_params = filter(lambda x: x.requires_grad, adjoint_params)
     adjoint_method = _select_default_adjoint_method(sde, adjoint_method)
+    if adjoint_options is None:
+        adjoint_options = {}
+    else:
+        adjoint_options = adjoint_options.copy()
 
-    ys = _SdeintAdjointMethod.apply(  # noqa
-        sde, ts, dt, bm, method, adjoint_method, adaptive, adjoint_adaptive, rtol, adjoint_rtol, atol,
-        adjoint_atol, dt_min, options, adjoint_options, y0, *adjoint_params
+    # Note that all of these warnings are only applicable for reversible solvers with sdeint_adjoint; none of them
+    # apply to sdeint.
+    try:
```

Comment:
Is such a try-except-else block the best? wouldn't something simpler like the following work
```python
if method == "rev_heun":
  # Check adjoint_method is expected_adjoint_method...
...
```

Just trying to keep things simple so that maintaining would be easy.